### PR TITLE
Radio Hotmic Noise + Alt Click Toggle

### PR DESCRIPTION
**Pull request by matttheficus**
## What Does This PR Do
Adds the radio transmit noise when a Radio hotmic is toggled on.
Allows you to toggle a radio's hotmic via Alt Click
Adds logging to prevent mass enabling of hotmics because people are stupid.

## Why It's Good For The Game
QoL upgrade for Handheld Radios mostly.

Also gives a heads up if someone activates a hotmic nearby.

Logging is due to people being stupid a lot of the time.

## Testing
Testing a buncha radios on local, seems to work.

## Changelog
:cl:
tweak: Radio hotmics can now be enabled via Alt Click + make a transmit noise on activation.
add: Added logging for hotkey hotmic enabling to prevent spam.
/:cl:

### CODE REVIEW

**Comment by Henri215** [comment]
No examination to tell the player he can toggle it with alt click?

**Comment by matttheficus** [comment]
> No examination to tell the player he can toggle it with alt click?

Guh, will add.

**Comment by Ionward** [comment]
if anyone didn't know either, you don't even need to hotmic a handheld if you just use :L or :R before your message, depending on which hand its in, if you're poking at examines that tell you controls might be worth making that known too

**Comment by Fordoxia** [comment]
> if anyone didn't know either, you don't even need to hotmic a handheld if you just use :L or :R before your message, depending on which hand its in, if you're poking at examines that tell you controls might be worth making that known too

Semi-related, I believe :i allows you to do the same with intercoms.

**Comment by procdrone** [comment]
> > if anyone didn't know either, you don't even need to hotmic a handheld if you just use :L or :R before your message, depending on which hand its in, if you're poking at examines that tell you controls might be worth making that known too
> 
> Semi-related, I believe :i allows you to do the same with intercoms.

Yes, it does

**Comment by matttheficus** [comment]
There is literally no changes to be made, bot. Rude!

File: code/__DEFINES/misc_defines.dm [hunk 1]
```diff
@@ -397,6 +397,8 @@
 
 #define INVESTIGATE_BOMB "bombs"
```
Comment (Burzah) [inline code/__DEFINES/misc_defines.dm:399]:
```suggestion
```

File: code/game/objects/items/devices/radio/radio_objects.dm [hunk 2]
```diff
@@ -252,6 +260,13 @@ GLOBAL_LIST_EMPTY(deadsay_radio_systems)
 
 /obj/item/radio/proc/ToggleBroadcast()
 	broadcasting = !broadcasting && !(wires.is_cut(WIRE_RADIO_TRANSMIT) || wires.is_cut(WIRE_RADIO_SIGNAL))
+	if(broadcasting)
+		playsound(src, 'sound/items/radio_common.ogg', rand(4, 16) * 5, SOUND_RANGE_SET(3) )
```
Comment (Burzah) [inline code/game/objects/items/devices/radio/radio_objects.dm:264]:
```suggestion
		playsound(src, 'sound/items/radio_common.ogg', rand(4, 16) * 5, SOUND_RANGE_SET(3))
```

**Review by Burzah — APPROVED** [review]

File: code/game/objects/items/devices/radio/radio_objects.dm [hunk 3]
```diff
@@ -252,6 +260,13 @@ GLOBAL_LIST_EMPTY(deadsay_radio_systems)
 
 /obj/item/radio/proc/ToggleBroadcast()
 	broadcasting = !broadcasting && !(wires.is_cut(WIRE_RADIO_TRANSMIT) || wires.is_cut(WIRE_RADIO_SIGNAL))
+	if(broadcasting)
+		playsound(src, 'sound/items/radio_common.ogg', rand(4, 16) * 5, SOUND_RANGE_SET(3))
+
+/obj/item/radio/proc/isBroadcasting()
+	if(broadcasting)
+		return TRUE
+	return FALSE
```
Comment (DGamerL) [inline code/game/objects/items/devices/radio/radio_objects.dm:269]:
```suggestion
	return broadcasting
```

File: code/game/objects/items/devices/radio/radio_objects.dm [hunk 4]
```diff
@@ -131,6 +131,14 @@ GLOBAL_LIST_EMPTY(deadsay_radio_systems)
 		return
 	ui_interact(user)
 
+/obj/item/radio/AltClick(mob/user)
+	if(user.stat || HAS_TRAIT(user, TRAIT_HANDS_BLOCKED) || !Adjacent(user) || !istype(user))
+		return
+
+	ToggleBroadcast()
+	to_chat(user, "<span class='notice'>You <b>[isBroadcasting() ? "enable" : "disable"]</b> [src]'s hotmic!</span>")
```
Comment (DGamerL) [inline code/game/objects/items/devices/radio/radio_objects.dm:139]: Why even use a proc for this? You could replace it with the var `broadcasting`

File: code/game/objects/items/devices/radio/radio_objects.dm [hunk 5]
```diff
@@ -252,6 +260,13 @@ GLOBAL_LIST_EMPTY(deadsay_radio_systems)
 
 /obj/item/radio/proc/ToggleBroadcast()
 	broadcasting = !broadcasting && !(wires.is_cut(WIRE_RADIO_TRANSMIT) || wires.is_cut(WIRE_RADIO_SIGNAL))
+	if(broadcasting)
+		playsound(src, 'sound/items/radio_common.ogg', rand(4, 16) * 5, SOUND_RANGE_SET(3))
+
+/obj/item/radio/proc/isBroadcasting()
```
Comment (DGamerL) [inline code/game/objects/items/devices/radio/radio_objects.dm:266]: Please make this `snake_case`

File: code/game/objects/items/devices/radio/radio_objects.dm (hunk 4 above)
Comment (matttheficus) [reply code/game/objects/items/devices/radio/radio_objects.dm:139]: Im a silly goose.

**Review by DGamerL — APPROVED** [review]

**Pull request merged**